src/prefect/events/clients.py

```diff
@@ -84,6 +84,20 @@ def events_out_socket_from_api_url(url: str):
     return http_to_ws(url) + "/events/out"
 
 
+def _host_matches_no_proxy(host: str) -> bool:
+    """Checks whether a given host matches NO_PROXY configuration.
+    NO_PROXY entries with a leading dot are treated as domain suffixes, while all others are treated as exact matches.
```

Comment:
I'm inclined to use `from urllib.request import proxy_bypass` here to avoid having our own implementation if possible.

My reading of [NO_PROXY](https://curl.se/libcurl/c/CURLOPT_NOPROXY.html) is that it's either an exact match or substring (as opposed to just a suffix), which would conflict with this as written. I'm not very familiar, so just want to compare notes if you saw something that suggested it should work differently.

Reply:
Was looking for precisely this, also not super familiar here.
This looks indeed perfect, swapped it out and working as expected.
Pushed a change to swap out just now